### PR TITLE
Add `CTestInfo` class, add test for create-close-remove process

### PR DESCRIPTION
**Pull request by heinrich5991**

### CODE REVIEW

File: src/test/fs.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#include <gtest/gtest.h>
+
+#include <base/system.h>
+
+TEST(Filesystem, CreateCloseDelete)
+{
+	CTestInfo Info;
+
+	IOHANDLE File = io_open(Info.m_aFilename, IOFLAG_WRITE);
+	ASSERT_TRUE(File);
```
Comment (def-) [inline src/test/fs.cpp]: Using true and false here seems a bit strange, (ab)using implicit conversions. what about ASSERT_THAT(File, NotNull()) and EXPECT_THAT(io_close(File), Eq(0))? THAT is a bit more modern and has fewer problems with compiler warnings in my experience.

File: src/base/system.c [hunk 2]
```diff
@@ -435,8 +435,7 @@ unsigned io_write_newline(IOHANDLE io)
 
 int io_close(IOHANDLE io)
 {
-	fclose((FILE*)io);
-	return 1;
+	return fclose((FILE*)io) != 0;
```
Comment (def-) [inline src/base/system.c]: Why not just return fclose((FILE*)io));?

Comment (heinrich5991) [reply src/base/system.c]: The rest of the file tries to normalize the return values to `0` and `1`, `!= 0` maps `0` to `0` and any other value to `1`.

File: src/test/fs.cpp (hunk 1 above)
Comment (heinrich5991) [reply src/test/fs.cpp]: Do you mean that I should use `EXPECT_TRUE(File != 0)`?

**Comment by heinrich5991** [comment]
I'll fix the other comment about `EXPECT_TRUE` if you tell me what you want instead. What about `EXPECT_TRUE(File != 0)`?

File: src/test/fs.cpp (hunk 1 above)
Comment (def-) [reply src/test/fs.cpp]: `EXPECT_THAT(File, NotNull());`

Comment (heinrich5991) [reply src/test/fs.cpp]:
```
../src/test/fs.cpp:11:20: error: ‘NotNull’ was not declared in this scope
  ASSERT_THAT(File, NotNull());
                    ^~~~~~~
../src/test/fs.cpp:11:2: error: ‘ASSERT_THAT’ was not declared in this scope
  ASSERT_THAT(File, NotNull());
  ^~~~~~~~~~~
```

Maybe `EXPECT_THAT` and `NotNull()` are from project-specific code in the code you know this from?

Comment (def-) [reply src/test/fs.cpp]: https://github.com/google/googletest/blob/master/googlemock/docs/CheatSheet.md#matchers

**Pull request merged**